### DISCUSSION TITLE
Using GeoJSON as tiles to display actively changeable POIs

### DISCUSSION THREAD

**Discussion by Roman-Kryvenchenko** [maplibre/maplibre-gl-js#3830] [Q&A] [answered]
Hi, I'm new in the map world so I'm seeking for an advice. 

I want to display a lot of points, in my case it'll be restaurants and cafes. 

I'm not sure what is the best approach here, but the goal is to request part of the json data from my API instead of loading it at once. Is there an existing solution for this case? 

I was thinking about adding data to my pmtiles with some tools, but in case when there are updates I'll need to rebuild it, which is not ideal. 

Thank you for reading, and hope you'll have an answer on my questions :D

**Comment by neodescis** [comment, marked as answer]
If you want to pull down data as the map is scrolling around, providing vector tiles is really the most performant solution, even if it is somewhat complex. You do not necessarily have to generate static tiles, though. For instance, we have used this repo as a basis to generate vector tiles on the fly using data from Elasticsearch: https://github.com/StationA/tilenol/tree/main

If you are looking for something less complex, you could create a geojson source and populate it (via `setData`) on the fly as the map scrolls around. This will be less performant, but then you can deal with GeoJSON directly and not have to worry about vector tiles.

**Reply by Roman-Kryvenchenko** [reply]
Thank you, I'll try to see if that tiles generation will work for me. 

Also about this approach with `setData`. Do you mean to check on map user movements and request this data in case it changed enough (and then update source)?

**Reply by neodescis** [reply]
Yes, exactly. If you use tiles, MapLibre will request them as needed while you're zooming and scrolling. If you want to use a geojson source with data that's limited to the current view area, it becomes your responsibility to filter and repopulate data as needed :-)

**Reply by Roman-Kryvenchenko** [reply]
Perfect, another reason to learn GO :D 
Ok, closing this discussion, thanks again
